examples/test_gfc.cpp

```diff
@@ -0,0 +1,441 @@
+   //    'ref_levels' to be the largest number that gives a final mesh with no
+   //    more than 10,000 elements.
+   {
+      for (int l = 0; l < ser_ref_levels; l++)
+      {
+         mesh->UniformRefinement();
+      }
+   }
+
+   // 6. Define a parallel mesh by a partitioning of the serial mesh. Refine
+   //    this mesh further in parallel to increase the resolution. Once the
+   //    parallel mesh is defined, the serial mesh can be deleted.
+   ParMesh *pmesh = new ParMesh(MPI_COMM_WORLD, *mesh);
+   delete mesh;
+   {
+      for (int l = 0; l < par_ref_levels; l++)
+      {
+         pmesh->UniformRefinement();
+      }
+   }
+
+   // 7. Define a parallel finite element space on the parallel mesh. Here we
+   //    use continuous Lagrange finite elements of the specified order. If
+   //    order < 1, we instead use an isoparametric/isogeometric space.
+   FiniteElementCollection *h1_fec;
+   FiniteElementCollection *dg_fec;
+   h1_fec = new H1_FECollection(order, dim);
+   dg_fec = new DG_FECollection(order, dim, BasisType::GaussLegendre,
+                                mtv ? FiniteElement::VALUE
+                                : FiniteElement::INTEGRAL);
+
+   ParFiniteElementSpace *h1_fespace = new ParFiniteElementSpace(pmesh, h1_fec);
+   ParFiniteElementSpace *dg_fespace = new ParFiniteElementSpace(pmesh, dg_fec);
+
+   ParFiniteElementSpace *fespace = dg ? dg_fespace : h1_fespace;
+   HYPRE_Int size = fespace->GlobalTrueVSize();
+   if (myid == 0)
+   {
+      cout << "Number of finite element unknowns: " << size << endl;
+   }
+
+   // 10. Define the solution vector x as a parallel finite element grid function
+   //     corresponding to fespace. Initialize x with initial guess of zero,
+   //     which satisfies the boundary conditions.
+   ParGridFunction x(fespace);
+   FunctionCoefficient coef(func);
+   x.ProjectCoefficient(coef);
+
+   GridFunctionCoefficient xCoef(&x);
+
+   double tol = 1e-6;
+   int npts = 0;
+
+   if (di)
+   {
+      // Testing Coefficient::Eval as it occurs in Bilinear- and LinearForm
+      // Domain Integrators and GridFunction::ProjectCoefficient.
+      cout << "Checking " << pmesh->GetNE()
+           << " elements in a non-DG context" << endl;
+      for (int i=0; i<pmesh->GetNE(); i++)
+      {
+         ElementTransformation *T = h1_fespace->GetElementTransformation(i);
+         const FiniteElement   *fe = h1_fespace->GetFE(i);
+         const IntegrationRule &ir = IntRules.Get(fe->GetGeomType(),
+                                                  2*order + 2);
+
+         double tip_data[3];
+         Vector tip(tip_data, 3);
+         for (int j=0; j<ir.GetNPoints(); j++)
+         {
+            npts++;
+            const IntegrationPoint &ip = ir.IntPoint(j);
+            T->SetIntPoint(&ip);
+            T->Transform(ip, tip);
+
+            double f_val = func(tip);
+            double gf_val = xCoef.Eval(*T, ip);
+
+            if (fabs(f_val - gf_val) > tol)
```

Comment:
Why is this written as an example as opposed to a unit test?

Reply:
This was written to share with the developer of PR #716 so that we could compare both approaches.  An example was the easiest option.  I would be happy to try and turn this into a unit test.